File: tests/fixtures/edgecase_repo/kotlin/Service2.kt
package com.example.gen

import com.example.gen.Base2

interface Handler2 {
    fun handle(payload: String): Boolean
}

open class Base2(val name: String)

class Service2(name: String) : Base2(name), Handler2 {
    override fun handle(payload: String): Boolean {
        validate(payload)
        return true
    }

    private fun validate(payload: String) {}
}

fun helper2(value: String): String {
    return value
}

File: tests/fixtures/edgecase_repo/csharp/Service2.cs
using System;

namespace Example.Gen
{
    public interface IHandler2
    {
        bool Handle(string payload);
    }

    public class Base2
    {
        protected string Name;
    }

    public class Service2 : Base2, IHandler2
    {
        public Service2(string name)
        {
            this.Name = name;
        }

        public bool Handle(string payload)
        {
            this.Validate(payload);
            return true;
        }

        private void Validate(string payload) { }
    }
}
